genesis-builder/src/generate.rs

```diff
@@ -84,9 +93,21 @@ pub fn l2_abi_to_move() -> anyhow::Result<()> {
                     ..Default::default()
                 };
 
-                if fun.state_mutability == StateMutability::Payable {
-                    function.has_value = true;
-                    has_fungible_asset = true;
+                match &fun.state_mutability {
+                    StateMutability::Payable => {
+                        has_evm_call = true;
+                        function.is_view = false;
```

Comment:
I believe if it's instantiated with `Default::default()`, only flipping to `true` is needed to be explicit. But it does aid readability, so mostly a nit.

Reply:
Yes, I like the explicit setting to `false` even though it's not strictly necessary. It may avoid problems in the future if we somehow accidentally change the behaviour of Default; it's quite unlikely of course, but the main point is that unchecked assumptions (such as assuming a `function` is marked as "not view" by default) can lead to bugs in future code changes, so it's safer to minimize the number of assumptions.